Add `Checkbox` functional tests.

lzhoucs opened the pull request:
**Type:** functional tests

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

<!--
Our bots should ensure:

* [ ] All contributors have signed a CLA
* [ ] The PR passes CI testing
* [ ] Code coverage is maintained
* [ ] The PR has been reviewed and approved
-->

Resolves #150

dylans commented on src/checkbox/tests/functional/Checkbox.ts:
```diff
@@ -0,0 +1,117 @@
+	'checkbox should be visible'(this: any) {
+		return getPage(this.remote)
+			.findByCssSelector(nthCheckbox(1))
+		// as per mwistrand, `isDisplayed` is broken in Safari 10, commented out for now
```
Same note here, let's either find an external issue or create one in meta for fixing in the future when Safari driver improves.

lzhoucs replied:
Issue link is added.

dylans commented on src/checkbox/tests/functional/Checkbox.ts:
```diff
@@ -0,0 +1,117 @@
+			})
+			.click()
+			.then(undefined, (err: Error) => {
+				// this then error callback is needed in FireFox with FirefoxDriver since it throws an
```
"this then error callback"?  Also link to a known issue please.

lzhoucs replied:
Issue link is added. Thanks!

lzhoucs commented:
@tomdye @dylans , thanks for the review. I've addressed your feedback. Please take another look when you get chance and let me know if there's any other issues. Thanks!
